src/services/SendToGovNotifyService.ts

```diff
@@ -0,0 +1,613 @@
+  				this.environmentVariables.getPdfEmailTemplateId(this.logger),
+  				message,
+  				options,
+  				clientConfig.GovNotifyApi,
+  			);
+
+  			await this.sendF2FYotiEmailedEvent(message);
+  			return emailResponse;
+  		} else {
+  			this.logger.error("Failed to fetch the Instructions pdf", {
+  				messageCode: MessageCodes.FAILED_FETCHING_YOTI_PDF,
+  			});
+  			throw new AppError(
+  				HttpCodesEnum.SERVER_ERROR,
+  				"sendYotiInstructions - Failed to fetch the Instructions pdf",
+  			);
+  		}
+  	} catch (err: any) {
+  		this.logger.error("sendYotiInstructions - Cannot send Email", {
+  			messageCode: MessageCodes.FAILED_TO_SEND_PDF_EMAIL,
+  		});
+  		throw new AppError(
+  			HttpCodesEnum.SERVER_ERROR,
+  			"sendYotiInstructions - Cannot send Email",
+  		);
+  	}
+  }
+
+  async fetchYotiPdf(sessionId: string, yotiSessionId: string): Promise<any> {
+  	
+  		const f2fSessionInfo = await this.f2fService.getSessionById(sessionId);
+
+  		if (!f2fSessionInfo) {
+  			this.logger.warn("Missing details in SESSION table", {
+  				messageCode: MessageCodes.SESSION_NOT_FOUND,
+  			});
+  			throw new AppError(HttpCodesEnum.BAD_REQUEST, "Missing details in SESSION table");
+  		}
+
+
+  	const streamToBuffer = async (stream: Readable): Promise<Buffer> => {
+  		return new Promise((resolve, reject) => {
+  			const chunks: Uint8Array[] = [];
+  			stream.on("data", (chunk) => chunks.push(chunk));
+  			stream.on("end", () => resolve(Buffer.concat(chunks)));
+  			stream.on("error", reject);
+  		});
+  	};
+		
+
+  		const s3Client = new S3Client({
+  			region: process.env.REGION,
+  			maxAttempts: 2,
+  			requestHandler: new NodeHttpHandler({
+  				connectionTimeout: 29000,
+  				socketTimeout: 29000,
+  			}),
+  		});
+  		const bucket = this.environmentVariables.yotiLetterBucket();
+  		const folder = this.environmentVariables.yotiPdfBucketFolder();
+  		const key = `${folder}/${yotiSessionId}`;
+		
+  		const pdfParams = {
+  			Bucket: bucket,
+  			Key: key,
+  		};
+
+  		this.logger.info("Fetching Yoti PDF from bucket");
+  	try {
+  		const yotiPdf = await s3Client.send(new GetObjectCommand(pdfParams));
+		  if (yotiPdf.Body instanceof Readable) {
+  			const body = await streamToBuffer(yotiPdf.Body);
+  			const encoded = body.toString("base64");
+  			return encoded;
+		  }
+  	} catch (error) {
+  		this.logger.error({ message: "Error fetching Yoti PDF from S3 bucket", error });
+  	}	
+  }
+
+  async fetchMergedPdf(sessionId: string, yotiSessionId: string): Promise<any> {
+  	const f2fSessionInfo = await this.f2fService.getSessionById(sessionId);
+  		if (!f2fSessionInfo) {
+  			this.logger.warn("Missing details in SESSION table", {
+  				messageCode: MessageCodes.SESSION_NOT_FOUND,
+  			});
+  			throw new AppError(HttpCodesEnum.BAD_REQUEST, "Missing details in SESSION table");
+  		}
+  		
+
+  	const streamToBuffer = async (stream: Readable): Promise<Buffer> => {
```

Comment:
Why duplicate the S3Client code duplication to fetch the pfd files in 2 methods? Why not extract it to a util file to b used in future and pass on only the required fields which are bucket name and key and have unit tests for that util seperatly. Avoids a lot of code duplication again here.